File: src/classes/api/req_res/update-ur-log-request.ts
'use strict'

import { URLog } from '@/classes/datas/ur-log'
import { GkillAPIRequest } from '../gkill-api-request'
import { Kyou } from '@/classes/datas/kyou'

export class UpdateURLogRequest extends GkillAPIRequest {

    urlog: URLog

    re_get_urlog_content: boolean

    tx_id: string | null = null

    want_response_kyou: boolean

    updated_kyou: Kyou | null | null = null

    constructor() {
        super()
        this.re_get_urlog_content = true
        this.urlog = new URLog()
        this.want_response_kyou = false
    }

}

File: src/app/gkill/api/req_res/update_ur_log_request.go
package req_res

import "github.com/mt3hr/gkill/src/app/gkill/dao/reps"

type UpdateURLogRequest struct {
	SessionID string `json:"session_id"`

	URLog *reps.URLog `json:"urlog"`

	ReGetURLogContent bool `json:"re_get_urlog_content"`

	TXID *string `json:"tx_id"`

	LocaleName string `json:"locale_name"`

	UpdatedKyou *reps.Kyou `json:"updated_kyou"`

	WantResponseKyou bool `json:"want_response_kyou"`
}
